pkg/providers/pricing/pricing.go

```diff
@@ -384,10 +384,10 @@ func (p *DefaultProvider) UpdateSpotPricing(ctx context.Context) error {
 		if _, ok := p.spotPrices[it]; !ok {
 			p.spotPrices[it] = newZonalPricing(0)
 		}
-		for zone, price := range zoneData {
+		for zone, price := range zoneData.prices {
```

Comment:
I think it's fine for now but we should come back to this since there's some re-assignment here that's done that we probably don't need to do with your new changes